src/Model/InteractiveCampaign.php

```diff
@@ -20,53 +21,90 @@ class InteractiveCampaign extends DataObject
 {
     private static $table_name = 'InteractiveCampaign';
 
-    private static $db = array(
+    private static $db = [
         'Title' => 'Varchar',
-        'Begins' => 'Date',
-        'Expires' => 'Date',
+        'Begins' => 'DBDatetime',
+        'Expires' => 'DBDatetime',
         'ResetStats' => 'Boolean',
         'DisplayType' => 'Varchar(64)',
         'TrackIn' => 'Varchar(64)',
         'AllowedHosts' => 'MultiValueField',
-    );
+    ];
 
-    private static $has_many = array(
+    private static $has_many = [
         'Interactives' => Interactive::class,
-    );
+    ];
 
-    private static $has_one = array(
+    private static $has_one = [
         'Client' => InteractiveClient::class,
-    );
+    ];
 
-    private static $extensions = array(
+    private static $extensions = [
         InteractiveLocationExtension::class,
         Versioned::class
-    );
+    ];
+
+    private static $datetimeFormat = 'Y-m-d H:i:00';
+
+    public function populateDefaults()
+    {
+        // begins
+        if ($begins = Config::inst()->get(self::class, 'Begins')) {
+            $this->Begins = date(static::$datetimeFormat, strtotime($begins));
+        } else {
+            $this->Begins = date(static::$datetimeFormat, strtotime('now'));
+        }
+        // expires
+        if ($expires = Config::inst()->get(self::class, 'Expires')) {
+            $this->Expires = date(static::$datetimeFormat, strtotime($expires));
+        } else {
+            // end of 30 days from now
+            $this->Expires = date(static::$datetimeFormat, strtotime('midnight + 31 days - 1 minute'));
+        }
+        // display type
+        if ($display_type = Config::inst()->get(self::class, 'DisplayType')) {
+            $this->DisplayType = $display_type;
+        }
+        // track in
+        if ($track_in = Config::inst()->get(self::class, 'TrackIn')) {
+            $this->TrackIn = $track_in;
+        }
+        parent::populateDefaults();
+    }
 
     public function getCMSFields()
     {
         $fields = parent::getCMSFields();
 
+        $dtFormat = 'Format: "dd/mm/yyyy, hh:mm"';
```

Comment:
This format is user-locale specific - you might be best setting it to YYYY-MM-DD H:i for simplicity

Reply:
I'm not 100% here but that's just an added description for the format that DBDatetime fields use by default. Actual values given to them are calculated like `$this->DBDatetime date('Y-m-d H:i:00', strtotime('now'))`. Is that user-local specific?

Reply:
Misleading field descriptions removed.